Mathlib/Topology/Order/DenselyOrdered.lean

```diff
@@ -229,6 +229,33 @@ theorem comap_coe_nhdsWithin_Iio_of_Ioo_subset (hb : s ⊆ Iio b)
     obtain ⟨x : s, hx : ∀ z, x ≤ z → z ∈ u⟩ := mem_atTop_sets.1 hu
     exact ⟨Ioo x b, Ioo_mem_nhdsWithin_Iio' (hb x.2), fun z hz => hx _ hz.1.le⟩
 
+/-- A set with a point in both the frontier and the set is not open -/
+private lemma frontier_to_not_IsOpen{α: Type*} [TopologicalSpace α] {s: Set α} {a: α}
+    (hb: a ∈ s) (hf: a ∈ frontier s): ¬IsOpen s := by
+  apply not_imp_not.mpr IsOpen.inter_frontier_eq
+  simp_rw [Set.eq_empty_iff_forall_not_mem, Set.mem_inter_iff, not_forall, Classical.not_not]
+  refine ⟨a, ⟨hb, hf⟩⟩
+
+/-- The interval `(-∞, a]` is not open. -/
```

Comment:
I'm not quite sure what the convention for punctuation marks in docstrings is, but I'd rather see all of this end in a period, or none of them.